pallets/corporate-actions/src/lib.rs

```diff
@@ -425,6 +433,40 @@ decl_module! {
             // Emit event.
             Self::deposit_event(Event::CAInitiated(caa, id, kind, record_date, details, targets, dwt, wt));
         }
+
+        /// Link the given CA `id` to the given `docs`.
+        /// Any previous links for the CA are removed in favor of `docs`.
+        ///
+        /// The workflow here is to add the documents and initiating the CA in any order desired.
+        /// Once both exist, they can now be linked together.
+        ///
+        /// ## Arguments
+        /// - `id` of the CA to associate with `docs`.
+        /// - `docs` to associate with the CA with `id`.
+        ///
+        /// # Errors
+        /// - `UnauthorizedAsAgent` if `origin` is not `ticker`'s sole CAA (owner is not necessarily the CAA).
+        /// - `NoSuchCA` if `id` does not identify an existing CA.
+        /// - `NoSuchDoc` if any of `docs` does not identify an existing document.
+        #[weight = <T as Trait>::WeightInfo::link_ca_doc(docs.len() as u32)]
+        pub fn link_ca_doc(origin, id: CAId, docs: Vec<DocumentId>) {
+            // Ensure that CAA is calling and that CA and the docs exists.
+            let caa = Self::ensure_ca_agent(origin, id.ticker)?;
+            Self::ensure_ca_exists(id)?;
+            for doc in &docs {
+                <Asset<T>>::ensure_doc_exists(&id.ticker, doc)?;
+            }
+
+            // TODO(Centril): Consider whether we want to dedup `docs` here.
+            // This will be considerably faster once `docs` is a vector of IDs.
+            // It would turn what is currently an `O(1)` operation in terms of
+            // local work (not storage reads!) into an `O(n log n)` operation.
+            // The benefits of deduping here are not entirely obvious.
```

Comment:
@maxsam4 please take a look at this.

Reply:
I wouldn't bother with it. I don't see any benefit of deduping it.